### ISSUE TITLE
Minor: Leading Spaces in XML Certificate Signature Algorithm

### ISSUE THREAD

**Issue by traffle-r7** [rbsec/sslscan#211]
While I was working on an XSL parser to convert the XML output into HTML tables, I noticed that the certificate signature algorithm contains some leading spaces (and a line break?):

```
<signature-algorithm>    Signature Algorithm: sha256WithRSAEncryption
</signature-algorithm>
```

**Comment by rbsec** [comment]
Hi @traffle-r7.

I'd actually noticed this at some point in the past and had meant to look into it, but never quite got round to it - so thanks for the reminder.

The weird formatting is because the code was using `X509_signature_print()` for some reason (not sure why, I think this is the original code), which adds the spacing. It also adds the unnecessary "Signature Algorithm:" text in the output, which should never really have been there.

I've updated this in the 2.0.0-beta5 tag. This is a potentially breaking change to the XML, which isn't something I like doing - but the certificate XML is already being changed for version 2.0.0, so I think it's fine here.

I'd be interested to see you XSL when it's done. There was a previous one that got made (I think by the person who did the Nmap one) that might be worth looking at - although it'll be a little outdated now:

https://gist.github.com/honze-net/6fc1e810aadcf153cde1a5fe99ff522e

Thanks

**Comment by traffle-r7** [comment]
I'll have to keep an eye out for those changes as they make their way to the apt-installed version on Kali and update accordingly, cool.

I'd just finished up my XSLT today, using some test results with 2.0.0-static on Kali: https://github.com/actuated/xsltsslscan/blob/master/xsltsslscan.xsl

It's definitely a bit more involved than that one, in part because this is my first stab at XSL stuff, but also just for the sake of everything I'm trying to pull while attempting to replicate the color-coded "bad" results.

I had an old shell script for converting the stdout results to HTML tables, but while I was getting requests to update it to include the additional info from sslscan 2.0, it seemed like a good time to revamp that entire process.

**Comment by rbsec** [comment]
I've had a quick play with it and it looks useful - I've never really done any XSLT stuff myself.

Good spot that the XML doesn't flag when servers accept any signature algorithms, I've fixed that in fefade5 (2.0.0-beta6).

**Comment by Sija** [comment]
Seems like this issue can be closed as it's already resolved since `2.0.0-beta6`.
